analyses/DESeq_analysis/run-DESeq-analysis.R

```diff
@@ -0,0 +1,195 @@
+# Function: 
+# 1. summarize Differential expression from RNASeq data
+# 2. tabulate corresponding P-value
+
+# Example run: DESeq
+# Rscript analyses/DESeq/run-DESeq-analysis.R 1 1
+# This was compares 1 cancer group (Combined or Cohort specific) with one GTEx tissue type. The input arguments are the indices of the comparison of the cancer groups and gtex sub tissues to be compared. In v6, there are 107 cancer groups vs 54 GTEx tissues
+
+# Read arguments from terminal
+args <- commandArgs(trailingOnly = TRUE)
+
+# Load required libraries
+suppressPackageStartupMessages({
+  library(ggplot2)
+  #library(DESeq2)
+})
+
+
+HIST_index <- args[1]   #assign first argument to Histology index
+GTEX_index <- args[2]   #assign second argument to GTEx index
+
+    
+
+
+
+#Load histology file
+hist <- read.delim("histologies.tsv", header=TRUE, sep = '\t')
+
+#Load expression counts data
+countData <- readRDS("gene-counts-rsem-expected_count-collapsed.rds")
+
+#Load expression TPM data
+TPMData <- readRDS("gene-expression-rsem-tpm-collapsed.rds")
+
+#Load EFO-MONDO map file
+EFO_MONDO <- read.delim("efo-mondo-map.tsv", header =T)
+
+#Load gene symbol-gene ID RMTL file
+ENSG_Hugo <- read.delim("ensg-hugo-rmtl-v1-mapping.tsv", header =T)
+
+# Subset Histology file for samples only found in the current the countData file (To ensure no discepancies cause errors later in the code)
+hist.filtered = unique(hist[which(hist$Kids_First_Biospecimen_ID %in%  colnames(countData)),])
+
+#Subset countadata for data that are present in the hitstoly files (To ensure no discepancies cause errors later in the code)
+countData_filtered = countData[,which(colnames(countData) %in% hist$Kids_First_Biospecimen_ID)]
+
+#Subset countadata for data that are present in the hitstoly files (To ensure no discepancies cause errors later in the code)
+TPMData_filtered = TPMData[,which(colnames(TPMData) %in% hist$Kids_First_Biospecimen_ID)]
+
+#Save all the unique cancer histologies in a variable. These cancer histologies represent the patient data in the countsdata
+Cancer_Histology <- unique(hist.filtered$cancer_group)
+
+#Save all the GTEx tissue subgroups in a variable. These cancer histologies represent the GTEx RNDA data available in the countsdata
+Gtex_Tissue_subgroup = sort(unique(hist.filtered$gtex_subgroup))
+
+#Save all the cohorts represented in the countsdata into a variable. Renove all 'NA's from the list. and paste cohort to cancer groep (eg GMKF_Neuroblastoma)
+Cancer_Histology_COHORT <- unique(paste(hist.filtered$cohort[which(!is.na(hist.filtered$cancer_group))],hist.filtered$cancer_group[which(!is.na(hist.filtered$cancer_group))],sep="_"))
+
+#Save all the histologies represented in the countsdata into a variable. Renove all 'NA's from the list. This will be the basis of all the data from each histology combined regardless of cohort (eg Combined_Neuroblastoma)
+Cancer_Histology <- paste("Combined",Cancer_Histology[which(!is.na(Cancer_Histology))],sep="_")
```

Comment:
We are currently using `all_cohorts` to denote the combined samples, which is noted at <https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/27#pullrequestreview-701391453>.

After changing `Combined` to `all_cohorts` the following `strsplit(..., split = '_')` statements also need to be revised to extract the correct part.

```suggestion
Cancer_Histology <- paste("all_cohorts",Cancer_Histology[which(!is.na(Cancer_Histology))],sep="_")
```